modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/java/com/liferay/frontend/data/set/admin/web/internal/model/listener/CompanyModelListener.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.frontend.data.set.admin.web.internal.model.listener;
+
+import com.liferay.frontend.data.set.admin.model.DataSetModelManager;
+import com.liferay.portal.kernel.exception.ModelListenerException;
+import com.liferay.portal.kernel.model.BaseModelListener;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.model.ModelListener;
+import com.liferay.portal.kernel.transaction.TransactionCommitCallbackUtil;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Daniel Sanz
+ */
+@Component(service = ModelListener.class)
+public class CompanyModelListener extends BaseModelListener<Company> {
+
+	@Override
+	public void onAfterCreate(Company company) throws ModelListenerException {
+		TransactionCommitCallbackUtil.registerCallback(
+			() -> {
+				_dataSetModelManager.checkCompany(company.getCompanyId());
```

Comment:
[naming nit] Looking in isolation here, it's unclear why we are checking companies after company creation. Maybe a more descriptive method name could be something like `.createObjectDefinitions()` or `.generateObjectDefinitions()`?

Reply:
Yep, I consider this naming temporary. Not yet clear if checkCompany() is the right name